src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp

```diff
@@ -81,16 +81,19 @@ namespace {
         const auto wat    = ph.active(::Opm::Phase::WATER);
         const auto oil    = ph.active(::Opm::Phase::OIL);
         const auto gas    = ph.active(::Opm::Phase::GAS);
+
         const auto threeP = gas && oil && wat;
+        const auto oilwater = !gas && oil && wat;
 
         const auto family1 =       // SGOF/SLGOF and/or SWOF
             (gas && (tm.hasTables("SGOF") || tm.hasTables("SLGOF"))) ||
             (wat && tm.hasTables("SWOF"));
+        // note: we allow for SOF2 to be part of family1 for threeP + solvent simulations.
 
         const auto family2 =      // SGFN, SOF{2,3}, SWFN
             (gas && tm.hasTables("SGFN")) ||
             (oil && ((threeP && tm.hasTables("SOF3")) ||
-                     tm.hasTables("SOF2"))) ||
+                     (oilwater && tm.hasTables("SOF2")))) ||
```

Comment:
I'm not sure I understand exactly what you're trying to do here.  If the run actually uses Family II this will check for `SOF2` only in an oil/water system.  If the run is oil/gas, then this check will fail.  What combinations of keywords are you trying to allow?

Reply:
I forgot that SOF2 is also used for 2p oil/gas systems. I will exchange oilwater with twoP.  What I want to allow is SWOF + SGOF + SOF2.